spec/shared_contexts/digests.rb

```diff
@@ -17,7 +17,11 @@
 
   def self.with_digest_algorithms(&block)
     DIGEST_ALGORITHMS_TO_TRY.each do |digest_algorithm|
-      describe("when digest_algorithm is #{digest_algorithm}", :digest_algorithm => digest_algorithm) do
+      describe(
+        "when digest_algorithm is #{digest_algorithm}",
+        {:digest_algorithm => digest_algorithm}.merge(
+          Puppet.features.microsoft_windows? && ['sha384', 'sha512'].include?(digest_algorithm) ? {:skip => "Windows Long File Name support is incomplete PUP-8257"} : {}
```

Comment:
Same here for `Puppet::Util::Platform.windows?`

Reply:
Actually, we don't need this at all if the existing skips are modified as suggested.

Reply:
It does cause passing tests to be skipped, but there isn't really a reasonable way to get at the `digest_algorithm` in the individual tests that are failing.